homeassistant/components/script/__init__.py

```diff
@@ -254,8 +225,51 @@ async def toggle_service(service):
     return True
 
 
-async def _async_process_config(hass, config, component):
-    """Process script configuration."""
+async def _async_process_config(hass, config, component) -> bool:
+    """Process script configuration.
+
+    Return true, if Blueprints were used.
+    """
+    entities = []
+    blueprints_used = False
+
+    for config_key in extract_domain_configs(config, DOMAIN):
+        LOGGER.error(config_key)
+        conf: dict[str, dict[str, Any] | BlueprintInputs] = config[config_key]
+
+        for object_id, config_block in conf.items():
+            raw_blueprint_inputs = None
+            raw_config = None
+
+            if isinstance(config_block, BlueprintInputs):
+                blueprints_used = True
+                blueprint_inputs = config_block
+                raw_blueprint_inputs = blueprint_inputs.config_with_inputs
+
+                try:
+                    raw_config = blueprint_inputs.async_substitute()
+                    config_block = cast(
+                        dict[str, Any],
+                        await async_validate_config_item(hass, raw_config),
+                    )
+                except vol.Invalid as err:
+                    LOGGER.error(
+                        "Blueprint %s generated invalid script with input %s: %s",
+                        blueprint_inputs.blueprint.name,
+                        blueprint_inputs.inputs,
+                        humanize_error(config_block, err),
+                    )
+                    continue
+            else:
+                raw_config = cast(ScriptConfig, config_block).raw_config
```

Comment:
This is more or less duplicated in automation/__init__.py, can it be moved to a shared blueprint helper?

Reply:
Dunno if this pattern will stick with other integrations in the future at this point.
Automations and scripts are much a-like, but still slightly different. Compared to e.g., the rest integrations, it differs more.

Reply:
Yeah let's not consolidate yet.